Fable.Remoting.ClientV2/Proxy.fs

```diff
@@ -146,8 +146,17 @@ module Proxy =
                             |> Http.sendAndReadBinary
                     
                     match statusCode with 
-                    | 200 -> 
-                        return unbox response 
+                    | 200 ->
+                        let rec getReturnType typ =
+                            if Reflection.FSharpType.IsFunction typ then
+                                let _, res = Reflection.FSharpType.GetFunctionElements typ
+                                getReturnType res
+                            elif typ.IsGenericType then
+                                typ.GetGenericArguments () |> Array.head
+                            else
+                                typ
+                        
+                        return MsgPack.Reader(response).Read (getReturnType fieldType)
```

Comment:
I don't like this bit here. `returnType` is created higher up in the function but it's just `TypeInfo` computed by (I think) SimpleJson with no reference to the actual `Type`. Is there a reason why `TypeInfo` does not contain it?

Reply:
Yeah this is indeed computed by SimpleJson. I think it doesn't expose the underlying `Type` because both SimpleJson and Remoting.Client primarily rely on the high-level `TypeInfo` representation of the types which is nicer to work with than the `Type` (at least that was the idea when I wrote that layer of abstraction) 

`TypeInfo` does expose the underlying type for some types (enums, unions and records) but I think you want the original type with as well

Reply:
Shall I leave this as it is for now?